### DISCUSSION TITLE
File Harvesting Error: SelfReg DLL and Assembly Issues

### DISCUSSION THREAD

**Discussion by SelaXT** [wixtoolset/issues#8196] [Questions]
Hi,

I'm having trouble implementing harvesting on a wix5 project. I tried implementing the way described in the documentation:
https://wixtoolset.org/docs/tools/heat/   
and also on this tutorial: https://www.youtube.com/watch?v=XcUmwegYClg

This is my MSI project code:
```
<Project Sdk="WixToolset.Sdk/5.0.0">
  <!--Harvest-->
  <ItemGroup>
    <HarvestDirectory Include="..\..\..\bin\Release">
      <ComponentGroupName>HarvestedComponents</ComponentGroupName>
      <DirectoryRefId>INSTALLFOLDER</DirectoryRefId>
      <SuppressRootDirectory>true</SuppressRootDirectory>
      <SupressRegistry>true</SupressRegistry>
      <SuppressCom>true</SuppressCom>
    </HarvestDirectory>
    
    <BindPath Include="..\..\..\bin\Release" />
  </ItemGroup>

    <!--Global Wix arguments-->
  <PropertyGroup>
    <VerboseOutput>True</VerboseOutput>
  </PropertyGroup>
  <!--Project specific properties-->
  <PropertyGroup>
    <ParamSourceBinDir>..\..\..\bin\Release</ParamSourceBinDir>
    <ParamSourceSeedDataReleaseDir>..\..\..\bin\Release\Data\SeedDataRelease</ParamSourceSeedDataReleaseDir>
    <OutputName>MyProject $(AssemblyVersion)</OutputName>
  </PropertyGroup>
  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|x64'">
    <OutputPath>bin\$(Configuration)\</OutputPath>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="WixToolset.Bal.wixext" Version="5.0.0" />
    <PackageReference Include="WixToolset.Heat" Version="5.0.0" />
    <PackageReference Include="WixToolset.Util.wixext" Version="5.0.0" />
  </ItemGroup>
</Project>

```

---
When I build the solution I get two types of errors:
1>heat.exe : warning HEAT5150: Could not harvest data from a file that was expected to be a SelfReg DLL: D:\GIT\MyProject \bin\Release\7za.exe. If this file does not support SelfReg you can ignore this warning. Otherwise, this error detail may be helpful to diagnose the failure: Unable to load file: D:\GIT\MyProject\bin\Release\7za.exe, error: 193

1>heat.exe : warning HEAT5151: Could not harvest data from a file that was expected to be an assembly: D:\GIT\MyProject \bin\Release\Autofac.Configuration.dll. If this file is not an assembly you can ignore this warning. Otherwise, this error detail may be helpful to diagnose the failure: Could not load file or assembly 'Autofac, Version=6.0.0.0, Culture=neutral

The last warning is shown for all the other DLLs.

Am I missing something? Should I add some other properties to the HarvestDirectory? Should I add something somewhere else?

**Comment by Thylon1** [comment]
In general if you are using Wix5 it might be better to use the new <Files> element for harvesting instead of heat.exe.
https://wixtoolset.org/docs/fivefour/#file-harvesting

Heat is more or less deprecated in newer Wix5+
